Initial Data Loading and Cleaning

In [15]:
#libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [16]:
#define relative path to dataset
path_train = '/Users/sumaiyauddin/Documents/ML/Project/Project Repo/sign_mnist_train.csv'
path_test = '/Users/sumaiyauddin/Documents/ML/Project/Project Repo/sign_mnist_test.csv'

#define image resolution 
res = (28,28)

In [17]:
#saving data into pandas dataframes 
test = pd.read_csv(path_test)
train = pd.read_csv(path_train)

In [18]:
#printing shape
print(f'Train Shape:{train.shape}')
print(f'Test Shape:{test.shape}')


Train Shape:(27455, 785)
Test Shape:(7172, 785)


The training dataset consists of 27455 28x28 pixel grayscale images, including the associated label. The testing dataset consists of 7172 28x28 images. The label and pixels are split into X and y for both training and testing data.

In [19]:
#training data 
X_train = train.iloc[:,1:]
y_train = train.iloc[:,0]

#testing data
X_test = test.iloc[:,1:]
y_test = test.iloc[:,0]

The below is a function to find the indices of a specified label.

In [20]:
#fuction to find the indices given a label 
def find_indices(data,label):
    #check if data is numpy array
    if type(data) == np.ndarray:
        #return indices
        return np.where(data==label)
    #check if data is pandas series 
    elif type(data) == pd.Series:
        #return indices
        return data[data==label].index
    #else not supported in this function
    else:
        raise Exception('Not supported data type for this function.')

Creating a dictionary where key is the numeric label in the dataset and value is the corresponding letter.

In [21]:
#letters
letters = ['A','B','C','D','E','F','G','H','I','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y']
#numbers 
numbers = sorted(list(y_train.unique()))
#dictionary of labels 
labels = dict(zip(numbers,letters))

Inspecting examples for each letter in the dataset.